src/NServiceBus.Core/Serializers/SystemJson/SystemJsonConfigurationExtensions.cs

```diff
@@ -0,0 +1,64 @@
+﻿namespace NServiceBus
+{
+    using System.Text.Json;
+    using NServiceBus.Configuration.AdvancedExtensibility;
+    using NServiceBus.Serialization;
+    using NServiceBus.Serializers.SystemJson;
+
+    /// <summary>
+    /// Extensions for <see cref="SerializationExtensions{T}"/> to manipulate how messages are serialized via System.Text.Json.
+    /// </summary>
+    public static class SystemJsonConfigurationExtensions
+    {
+        /// <summary>
+        /// Configures the <see cref="JsonReaderOptions"/> to use.
+        /// </summary>
+        /// <param name="config">The <see cref="SerializationExtensions{T}"/> instance.</param>
+        /// <param name="options">The <see cref="JsonReaderOptions"/> to use.</param>
+        public static void ReaderOptions(this SerializationExtensions<SystemJsonSerializer> config, JsonReaderOptions options)
+        {
+            var settings = config.GetSettings();
+            settings.Set(options);
```

Comment:
I would suggest to either use a serializer specific options type that we store and retrieve settings from our add a unique key. 

For example

```csharp
class SystemJsonSettings {
   public JsonReaderOptions ReaderOptions { get; set; }
   ...
}
```

or `settings.Set("SystemJsonSerializer.ReaderOptions", options)` or use the common root defined in https://github.com/Particular/NServiceBus/pull/6749/files#diff-e9ee8c710186c78173f2b1fef025ba7124d7817d3be72c7ef79438ab7202f797R62

This avoids problems with downstreams like persisters that also try to store system text json settings, or it would even allow a downstream to take out the message serializer options and share it with the persistence options (json options can clone from other options)